projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/case-flag/components/case-flag-summary-list/case-flag-summary-list.component.ts

```diff
@@ -47,16 +52,32 @@ export class CaseFlagSummaryListComponent implements OnInit {
     return `${flagDetail.name}${otherDescription}${subTypeValue}`;
   }
 
-  private getHeaderText(): string {
+  private getAddUpdateFlagHeaderText(): string {
+    if (this.displayContextParameter === CaseFlagDisplayContextParameter.CREATE) {
```

Comment:
@johnbenjamin-hmcts - there are quite a few `if` conditions... could the whole thing be rewritten as a `switch()` statement perhaps?

Reply:
@Daniel-Lam - I have done the changes as you suggested.